pkg/daemon/ceph/osd/agent.go

```diff
@@ -218,6 +232,30 @@ func (a *OsdAgent) configureDevices(context *clusterd.Context, devices *DeviceOs
 	return osds, nil
 }
 
+func getDeviceLVPath(context *clusterd.Context, deviceName string) (string, error) {
+	var lvPath string
+	cmd := fmt.Sprintf("get logical volume path for device")
+	output, err := context.Executor.ExecuteCommandWithOutput(false, cmd, "pvdisplay", "-C", "-o", "lvpath", deviceName, "--reportformat", "json")
+	if err != nil {
+		return "", fmt.Errorf("failed to retrieve logical volume path. %+v", err)
+	}
+
+	logger.Debugf("logical volume output : %+v", output)
+	var lvPathResult map[string][]map[string][]LVMInfo
```

Comment:
Have you tried looking with udev?

Reply:
@leseb  The input that I used here is in the format `/mnt/<device name>` and  `udev` only works for `/dev` directory. More details in this [comment](https://github.com/rook/rook/pull/3996/#discussion_r336932132)

Reply:
Using `--reportformat json` seems a bit overkill since `pvdisplay -C -o lvpath --noheadings /mnt/vdb` does the job too.

```
# pvdisplay -C -o lvpath --noheadings /mnt/vdb
  /dev/ceph-b5c9827c-2933-43f1-b5bc-9ca79ef5f045/osd-data-b24966a5-ded4-43e3-acc8-ee7d73a955eb
```

Let's not overcomplicate things.

Reply:
Thanks for this suggestion @leseb . I've updated the code to use this way to get the lvpath.